### PR TITLE
Fix: Pick the meta object from actual resource rather from cached res…

### PR DESCRIPTION
**Pull request by mountainfirefly**
…ource if it is different

### CODE REVIEW

**Comment by wopian** [comment]
Please can you fix the linter errors and warnings in the PR. You can run `yarn lint --fix` and it'll do most of it for you 👍

File: packages/kitsu-core/src/linkRelationships/index.js [hunk 1]
```diff
@@ -67,7 +67,23 @@ function linkObject (data, included, key, previouslyLinked, relationshipCache) {
   data[key] = {}
   const resource = data.relationships[key].data
   const cache = previouslyLinked[`${resource.type}#${resource.id}`]
-  data[key].data = cache || link(resource, included, previouslyLinked, relationshipCache)
+
+  if (cache) {
+    let resourceCache = null;
+    // Comparing for cache entity meta and resource entity meta object.
+    if (!isDeepEqual(cache.meta, resource.meta)) {
```
Comment (wopian) [inline packages/kitsu-core/src/linkRelationships/index.js:74]: Function is missing an import

File: packages/kitsu-core/src/deepEqual/index.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+ * @param {object} object Check if passed param is an object
+ * @returns {boolean} A boolean that indicates if passed param is an object
+ *
+ * @example <caption>Check for object</caption>
+ * isObject({
+ *   "firstName": "John",
+ *   "lastName": "Doe",
+ *   "age": 35 
+ * }) // true
+ */
```
Comment (wopian) [inline packages/kitsu-core/src/deepEqual/index.js:55]: Function needs a `@private` flag at the end of the JSDoc so the non-exported function isn't rendered on the README docs

File: packages/kitsu-core/src/deepEqual/index.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    }
+  }
+
+  return true;
+};
+
+/**
+ * Check for Object
+ *
+ * @name isObject
```
Comment (wopian) [inline packages/kitsu-core/src/deepEqual/index.js:45]: Name is not required, it is inferred from the function name by the documentation tooling

File: packages/kitsu-core/src/deepEqual/index.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
+/**
+ * Compare two objects equality 
+ *
+ * @name isDeepEqual
```
Comment (wopian) [inline packages/kitsu-core/src/deepEqual/index.js:4]: Name is not required as it is inferred from the function name by the documentation tooling

File: packages/kitsu-core/src/deepEqual/index.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+ * @param {object} firstObject First Object
+ * @param {object} secondObject Second Object to compare with the first one
+ * @returns {boolean} A boolean that indicates if objects are equal
+ *
+ * @example <caption>Deep equality check</caption>
+ * isDeepEqual({
+ *   "firstName": "John",
```
Comment (wopian) [inline packages/kitsu-core/src/deepEqual/index.js:11]: ESLint will have fixed the usages of this in the test file, but it won't touch these comments.

Can you update all of these to match the projects code style. No quoted property names and single quote strings (i.e `firstName: 'John'`)

File: packages/kitsu-core/src/deepEqual/index.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+ *   "lastName": "Doe",
+ *   "age": 35 
+}, {
```
Comment (wopian) [inline packages/kitsu-core/src/deepEqual/index.js:14]: Missing indentation

File: packages/kitsu-core/src/deepEqual/index.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+ *   "age": 35,
+ * }) // true
+ */
+export const isDeepEqual = (object1, object2) => {
+  const objKeys1 = Object.keys(object1);
+  const objKeys2 = Object.keys(object2);
+
+  if (objKeys1.length !== objKeys2.length) return false;
+
+  for (var key of objKeys1) {
```
Comment (wopian) [inline packages/kitsu-core/src/deepEqual/index.js:26]: `const key` as variable not reassigned

**Review by wopian — CHANGES_REQUESTED** [review]

**Comment by mountainfirefly** [comment]
Sure, will resolve these and update you soon.
Thanks

File: packages/kitsu-core/src/deepEqual/index.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
+/**
+ * Compare two objects equality 
+ *
+ * @param {object} firstObject First Object
+ * @param {object} secondObject Second Object to compare with the first one
+ * @returns {boolean} A boolean that indicates if objects are equal
+ * @private
```
Comment (wopian) [inline packages/kitsu-core/src/deepEqual/index.js:7]: Exported functions shouldn't have a private flag

**Review by wopian — CHANGES_REQUESTED** [review]
Please run `yarn lint --fix` to resolve the lining errors and warnings from ESLint

File: packages/kitsu-core/src/deepEqual/index.js [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+ *   age: 35
+ * },{
+ *   firstName: 'John',
+ *   lastName: 'Doe',
+ *   age: 35
+ * }) // true
+ */
+export const isDeepEqual = (object1, object2) => {
+  const objKeys1 = Object.keys(object1);
+  const objKeys2 = Object.keys(object2);
```
Comment (wopian) [inline packages/kitsu-core/src/deepEqual/index.js:21]: Function should check both values are objects before contining to access `Object.keys` on them. 

Tests are failing due to `undefined` and `null` values being passed in through `object1` and/or `object2`.

**Review by wopian — CHANGES_REQUESTED** [review]

**Comment by mountainfirefly** [comment]
@wopian Your review comments have been addressed, please check now.

Thanks

**Review by wopian — APPROVED** [review]

**Comment by wopian** [comment]
Thank you. I updated the tests to include coverage of the uncovered branches and added the sample you gave in the issue as a test case in `linkedRelationships` to test the behaviour works as intended.

**Comment by wopian** [comment]
Released in [10.0.3 (2022-10-30)](https://github.com/wopian/kitsu/releases/tag/v10.0.3)

**Comment by mountainfirefly** [comment]
Thank you so much @wopian .

**Pull request merged**